example/forest-example.cpp

```diff
@@ -1,12 +1,67 @@
 #include <forest/forest.hpp>
+#include <cstdio>
+#include <functional>
 #include <iostream>
+#include <iterator>
 
 constexpr auto foo = forest::literal<64>("<rgb=500><b><i>hello</b></i></rgb> world");
 
+void with_temp_file(std::function<void(std::FILE*)> const fn) {
```

Comment:
Use a template parameter `typename F` instead of `std::function`? It's quite heavy and mostly unnecessary here.

Reply:
Should I add an `std::is_invocable_v` check as well or no?

Reply:
Should I add an `std::is_invocable_v` check as well?

Reply:
Up to you! Since this is example "client side" code, it's not very important; different story if it were a library function.